client/image_inspect.go

```diff
@@ -5,28 +5,80 @@ import (
 	"context"
 	"encoding/json"
 	"io"
+	"net/url"
 
 	"github.com/docker/docker/api/types/image"
 )
 
-// ImageInspectWithRaw returns the image information and its raw representation.
-func (cli *Client) ImageInspectWithRaw(ctx context.Context, imageID string) (image.InspectResponse, []byte, error) {
+type imageInspectClientOpt struct {
+	raw        *bytes.Buffer
+	apiOptions image.InspectOptions
+}
+
+type ImageInspectOpt func(*imageInspectClientOpt)
+
+// ImageInspectWithRawResponse instructs the client to additionally store the
+// raw inspect response in the provided buffer.
+func ImageInspectWithRawResponse(raw *bytes.Buffer) ImageInspectOpt {
+	return func(opts *imageInspectClientOpt) {
+		opts.raw = raw
+	}
+}
+
+// ImageInspectWithOpts sets the API options for the image inspect operation.
+func ImageInspectWithOpts(opts image.InspectOptions) ImageInspectOpt {
+	return func(clientOpts *imageInspectClientOpt) {
+		clientOpts.apiOptions = opts
+	}
+}
+
+// ImageInspect returns the image information.
+func (cli *Client) ImageInspect(ctx context.Context, imageID string, inspectOpts ...ImageInspectOpt) (image.InspectResponse, error) {
```

Comment:
I can't say I fully like it. The "raw response" part is strictly a client feature, so it doesn't make sense to put it directly in the `image.InspectOptions`, on the other hand that forces us to have a client specific opts..

Any thoughts on this one? Would be nice to think this carefully, as it breaks the client interface.

Reply:
The linter will complain about the usage of deprecated `ImageInspectWithRaw`, but for now I'll refrain from adjusting these in case we reach a different conclusion on the shape of this.

Reply:
Hm;  I'll have a look at this one!

(For the linter; if there's  too many places to suppress it,  we  can also add a global exception in `.golangci-lint.yaml`)

Reply:
Actually I can just break the `Deprecated` comment for now to make the linter happy 😅